.github/workflows/rust.yml

```diff
@@ -20,3 +20,23 @@ jobs:
       run: cargo build --verbose
     - name: Run tests
       run: cargo test --verbose
+
+
+  coverage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust
+        run: rustup update stable
+      - name: Install cargo-llm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+      - name: Generate Code Coverage
+        run: cargo llvm-cov --all-features --workspace --lcov --output-path lcov.info
+      - name: Upload coverage to CodeCov
```

Comment:
Why upload to a third-party service? Seeing the results locally and in the PR build is enough, no?

Reply:
I see it shows graphs and history. Not sure how useful the history info will be. Aren't the graphs available for local runs?

Reply:
The main usage is for review. It will show which parts of the code is covered by code and which are not. So, for anyone reviewing they can see without they need to download the code and execute.

Graphs and history just nice add-ons. 

Local code coverage when we generate the html report, we will get the line by line coverage information. It does not have the graphs.